RecoTracker/TkDetLayers/src/GeometricSearchTrackerBuilder.cc

```diff
@@ -46,149 +42,120 @@ GeometricSearchTracker *GeometricSearchTrackerBuilder::build(const GeometricDet
   bool useBrothers = !usePhase2Stacks;
 
   vector<const GeometricDet *> theGeometricDetLayers = theGeometricTracker->components();
-  for (vector<const GeometricDet *>::const_iterator it = theGeometricDetLayers.begin();
-       it != theGeometricDetLayers.end();
-       it++) {
-    if ((*it)->type() == GeometricDet::PixelBarrel) {
-      vector<const GeometricDet *> thePxlBarGeometricDetLayers = (*it)->components();
-      for (vector<const GeometricDet *>::const_iterator it2 = thePxlBarGeometricDetLayers.begin();
-           it2 != thePxlBarGeometricDetLayers.end();
-           it2++) {
-        thePxlBarLayers.push_back(aPixelBarrelLayerBuilder.build(*it2, theGeomDetGeometry));
+  for (const GeometricDet *theGeomDetLayer : theGeometricDetLayers) {
+    if (theGeomDetLayer->type() == GeometricDet::PixelBarrel) {
+      vector<const GeometricDet *> thePxlBarGeometricDetLayers = theGeomDetLayer->components();
+      for (const GeometricDet *thisGeomDet : thePxlBarGeometricDetLayers) {
+        thePxlBarLayers.push_back(aPixelBarrelLayerBuilder.build(thisGeomDet, theGeomDetGeometry));
       }
     }
 
-    if ((*it)->type() == GeometricDet::PixelPhase1Barrel) {
-      vector<const GeometricDet *> thePxlBarGeometricDetLayers = (*it)->components();
-      for (vector<const GeometricDet *>::const_iterator it2 = thePxlBarGeometricDetLayers.begin();
-           it2 != thePxlBarGeometricDetLayers.end();
-           it2++) {
-        thePxlBarLayers.push_back(aPixelBarrelLayerBuilder.build(*it2, theGeomDetGeometry));
+    if (theGeomDetLayer->type() == GeometricDet::PixelPhase1Barrel) {
+      vector<const GeometricDet *> thePxlBarGeometricDetLayers = theGeomDetLayer->components();
```

Comment:
```auto const &```   ?

Reply:
I'd personally prefer to keep the original explicit declaration as it was before (here and in the other similar cases that you pointed out before): do you have any issue with it? You are the code expert, and I'll follow your prescription in that case

Reply:
try to add ```const &``` anyhow. Should help the compiler not to copy the vector!

Btw modenrize (in my opinion) is also use ```auto``` when is not confusing